contracts/Wallet.sol

```diff
@@ -15,54 +16,87 @@ contract Wallet is IWallet {
-    external view override returns (TokenInfo[] memory tokensInfo) {
-        // Get ERC-20 info
-        address[] memory erc20Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC20, start, limit);
-        // Get ERC-721 token info
-        address[] memory erc721Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC721, start, limit);
-        // Get ERC-1155 token info
-        address[] memory erc1155Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC1155, start, limit);
-
-        // Count total of the token
-        uint256 totalTokens = erc20Tokens.length + erc721Tokens.length + erc1155Tokens.length;
-        TokenInfo[] memory paginatedInfo = new TokenInfo[](totalTokens);
-        uint256 count = 0;
+    // Make a private method to fetch token info
+    function getTokenInfo(address tokenAddress, TokenType tokenType, address user)
+    private view returns (TokenInfo memory) {
+        TokenInfo memory info;
 
         // Fill in ERC-20 token information
-        for (uint256 i = 0; i < erc20Tokens.length; i++) {
-            IExtendedERC20 token = IExtendedERC20(erc20Tokens[i]);
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc20Tokens[i],
+        if (tokenType == TokenType.ERC20) {
+            IERC20Metadata token = IERC20Metadata(tokenAddress);
+            uint256 balance = token.balanceOf(user);
+
+            // Filter tokens with a balance of 0 in ERC-20
+            if (balance > 0) {
+                info = TokenInfo({
+                    tokenAddress: tokenAddress,
+                    balance: token.balanceOf(user),
+                    name: token.name(),
+                    symbol: token.symbol(),
+                    decimals: token.decimals()
+                });
+            }
+
+        // Fill in ERC-721 token information
+        } else if (tokenType == TokenType.ERC721) {
+            IERC721Metadata token = IERC721Metadata(tokenAddress);
+            info = TokenInfo({
+                tokenAddress: tokenAddress,
                 balance: token.balanceOf(user),
                 name: token.name(),
                 symbol: token.symbol(),
-                decimals: token.decimals()
+                decimals: 0 // ERC721 tokens do not have decimals
             });
-        }
 
-        // Fill in ERC-721 token information
-        for (uint256 i = 0; i < erc721Tokens.length; i++) {
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc721Tokens[i],
-                balance: 0,
-                name: "",
+        // Fill in ERC-1155 token information
+        } else if (tokenType == TokenType.ERC1155) {
+            IERC1155 token = IERC1155(tokenAddress);
+            // The token ID is needed to get the correct balance
+            // Assuming a token ID is provided or known
+            uint256 tokenId = 1; // Token ID needs to be determined, 1 is an example
```

Comment:
hardcoded tokenId ? You may refer to enumerable interface for ERC1155.

Reply:
Solved